wrapper/pre_copy.py

```diff
@@ -716,7 +763,8 @@ def _start_nbdkits(self, final):
                 filters.remove(filt)
 
         for disk in self.disks:
-            cmd = self._get_nbdkit_cmd(disk, filters, final)
+            cmd = getattr(self, "_get_nbdkit_cmd_%s" % self._transport_method)(
+                    disk, filters, final)
```

Comment:
Passing the transport method as a parameter should be enough and slightly more readable, I think. Actually, you already have it saved in the object already.